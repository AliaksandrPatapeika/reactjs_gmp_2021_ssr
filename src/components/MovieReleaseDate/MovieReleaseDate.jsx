import './MovieReleaseDate.less';

import PropTypes from 'prop-types';
import React from 'react';

const MovieReleaseDate = ({releaseDate}) => <div className="movieReleaseDate" title={`Release date: ${releaseDate}`}>{new Date(releaseDate).getFullYear()}</div>;

MovieReleaseDate.propTypes = {
  releaseDate: PropTypes.string.isRequired
};

export default MovieReleaseDate;
