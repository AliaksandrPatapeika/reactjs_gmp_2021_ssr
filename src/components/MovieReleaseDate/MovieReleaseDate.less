@import "../../less/variables";

.movieReleaseDate {
	padding: 1px 12px;
	border: 1.5px solid @gray;
	border-radius: 4px;
	font-size: 15px;
}
