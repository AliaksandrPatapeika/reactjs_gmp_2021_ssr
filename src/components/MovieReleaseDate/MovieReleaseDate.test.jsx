import {head} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import MockMovies from '../../tests/mocks/mockData';
import MovieReleaseDate from './MovieReleaseDate';

describe('MovieReleaseDate', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const movieReleaseDate = renderer
      .create(<MovieReleaseDate releaseDate={mockMovie.release_date} />)
      .toJSON();

    expect(movieReleaseDate).toMatchSnapshot();
  });
});
