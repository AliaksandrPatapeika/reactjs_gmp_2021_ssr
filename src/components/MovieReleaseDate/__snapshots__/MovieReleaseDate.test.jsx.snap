// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReleaseDate renders correctly 1`] = `
<div
  className="movieReleaseDate"
  title="Release date: 2004-03-15"
>
  2004
</div>
`;
